tap_ftp_files/__init__.py

```diff
@@ -43,15 +43,20 @@ def parse_args():
 def download(args):
     logger.debug(f"Downloading data...")
     config = args.config
-    host = config.get('host')
+    state = args.state or dict()
     target_dir = config.get('target_dir')
     file_groups = config.get('file_groups')
     incremental_mode = config.get('incremental_mode') == True
 
     conn = connection(config)
-    
-    start_date = config.get('start_date') if incremental_mode else None
-    
+    if incremental_mode:
+        if state.get('start_date'):
+            start_date = state.get('start_date')
+        elif incremental_mode:
```

Comment:
Why is this an `elif`? We already established `incremental_mode` is true on line 52